core/src/Instance/B2CAuthority.cs

```diff
@@ -68,16 +68,28 @@ internal B2CAuthority(string authority, bool validateAuthority)
                 throw new ArgumentException(CoreErrorMessages.UnsupportedAuthorityValidation);
             }
 
-            var metadata = await AadInstanceDiscovery
+            InstanceDiscoveryMetadataEntry metadata = await GetB2CInstanceDiscoveryMetadataEntryAsync(
```

Comment:
>GetB2CInstanceDiscoveryMetadataEntryAsync [](start = 60, length = 41)

Thinking the instance discovery point should have a virtual method or alternatively that the instance discovery object should be injected....

Reply:
Currently there is only one AADInstanceDiscovery. This is a singleton that maintains a cache. The B2CAuthority is a type of AADAuthority so it somewhat seems ok to be able to call AADInstanceDiscovery.

Reply:
outdated due to latest changes.

---
In reply to: [233710236](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1405#discussion_r233710236) [](ancestors = 233710236)

Reply:
outdated due to latest changes.

---
In reply to: [234182155](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1405#discussion_r234182155) [](ancestors = 234182155)